File: JDApp-Java/setting/src/main/java/com/jd/setting/application/SettingApplication.java
package com.jd.setting.application;

import android.app.Application;
import android.util.Log;

import com.renny.mylibrary.IAppInit;

/*
    @see https://github.com/ren93/initiator
 */

public class SettingApplication implements IAppInit {
    @Override
    public void init(Application application) {
        Log.d("init==", "HomeApplication");
    }
}

File: JDApp-Kotlin/setting/src/main/java/com/jd/setting/application/SettingApplication.kt
package com.jd.setting.application

import android.app.Application
import android.content.Context
import android.util.Log
import com.google.auto.service.AutoService
import com.jd.core.base.IApplication


@AutoService(IApplication::class)
class SettingApplication : IApplication {
    override fun onAttachBaseContext(context: Context) {

    }

    override fun onCreate(application: Application) {
        Log.d("init==", "SettingApplication")
    }

    override fun onTerminate(application: Application) {

    }

    override fun onAsyncInit() {

    }

}
